Marlin/Marlin_main.cpp

```diff
@@ -3794,7 +3807,38 @@ inline void gcode_G92() {
     sync_plan_position_e();
 }
 
-#if ENABLED(ULTIPANEL)
+#if ENABLED(ULTIPANEL) || (ENABLED(ULTRA_LCD) && ENABLED(CONTINUE_BUTTON_FEATURE) && PIN_EXISTS(CONTINUE)) || (ENABLED(CONTINUE_BUTTON_FEATURE) && PIN_EXISTS(CONTINUE) && PIN_EXISTS(LED))
+// We need an input and an output.
+  void wait_for_user(millis_t ms) {
+    KEEPALIVE_STATE(PAUSED_FOR_USER);
+
+    ms += (ms > 0) ? previous_cmd_ms : LONG_MAX;  // wait until this time for a click
+    #if PIN_EXISTS(BEEPER)    // Some boards have onboard beepers
+      buzzer.tone(1000, 300);
+    #endif
+    #if PIN_EXISTS(LED)
+      OUT_WRITE(LED_PIN, 1);
+    #endif
+
+    #if ENABLED(ULTIPANEL)
+      if (!lcd_detected()) return;
+      lcd_ignore_click(false);
+      while (PENDING(millis(), ms) && !lcd_clicked()) idle();
+	  lcd_ignore_click();
+    #elif ENABLED(CONTINUE_BUTTON_FEATURE) && PIN_EXISTS(CONTINUE)
```

Comment:
If `PIN_EXISTS` is sanity-checked for `CONTINUE_BUTTON_FEATURE` then we don't have to check `PIN_EXISTS(CONTINUE)` here or elsewhere.